tardis/montecarlo/montecarlo_numba/base.py

```diff
@@ -178,22 +160,15 @@ def montecarlo_main_loop(
     virtual_packet_logging : bool
         Option to enable virtual packet logging.
     """
-    output_nus = np.empty_like(packet_collection.packets_input_nu)
-    last_interaction_types = (
-        np.ones_like(packet_collection.packets_output_nu, dtype=np.int64) * -1
-    )
-    output_energies = np.empty_like(packet_collection.packets_output_nu)
+    no_of_packets = len(packet_collection.initial_nus)
+    output_nus = np.empty(no_of_packets, dtype=np.float64)
+    output_energies = np.empty(no_of_packets, dtype=np.float64)
 
-    last_interaction_in_nus = np.empty_like(packet_collection.packets_output_nu)
-    last_line_interaction_in_ids = (
-        np.ones_like(packet_collection.packets_output_nu, dtype=np.int64) * -1
-    )
-    last_line_interaction_out_ids = (
-        np.ones_like(packet_collection.packets_output_nu, dtype=np.int64) * -1
-    )
-    last_line_interaction_shell_ids = (
-        np.ones_like(packet_collection.packets_output_nu, dtype=np.int64) * -1
-    )
+    last_interaction_in_nus = np.empty(no_of_packets, dtype=np.float64)
+    last_interaction_types = -np.ones(no_of_packets, dtype=np.int64)
```

Comment:
Putting the `-` before the `np.ones()` call is a little difficult to read. I would prefer `-1 * np.ones()`

Reply:
done